app/controllers/katello/api/v2/organizations_controller.rb

```diff
@@ -68,14 +68,16 @@ def update
     param :organization, Hash do
       param :label, String, :required => false
     end
+    param :simple_content_access, :bool, :desc => N_('Whether to turn on Simple Content Access for the organization.'), :required => false, :default => true
     def create
       @organization = Organization.new(resource_params)
-      creator = ::Katello::OrganizationCreator.new(@organization)
-      creator.create!
-      @organization.reload
       # @taxonomy instance variable is necessary for foreman side
       # app/views/api/v2/taxonomies/show.json.rabl is using it.
       @taxonomy = @organization
+      sca = params.key?(:simple_content_access) ? ::Foreman::Cast.to_bool(params[:simple_content_access]) : true
+      creator = ::Katello::OrganizationCreator.new(@taxonomy, sca: sca)
+      creator.create!
```

Comment:
Moving `creator.create!` after `@taxonomy = @organization` is what fixed all the organizations_controller tests. (The problem was that `@taxonomy` was nil for some reason, so `creator.create!` was trying to create based on nothing.)  I don't see any harm in assigning `@taxonomy` earlier, but I may not fully understand why it was that way before. @parthaa @sjha4 thoughts?

Reply:
Not sure but the earlier code was calling reator = ::Katello::OrganizationCreator.new(@organization) and then @organization.reload upon create before assigning to taxonomy..

I don't know enough to opine what issues that might cause.

If this works, you'll be able to maintain the old flow:

creator = ::Katello::OrganizationCreator.new(@organization, sca: sca)

Reply:
you're right, that's probably better! Will update